code/src/componets/Form.js

```diff
@@ -0,0 +1,215 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable react/jsx-closing-tag-location */
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/jsx-closing-bracket-location */
+import React, { useState } from 'react';
+import headerImgSmall from '../assets/DSG-small.jpg';
+import headerImgMedium from '../assets/DSG-medium.jpg';
+import headerImgLarge from '../assets/DSG-large.jpg';
+import './Form.css';
+
+const ProgressBar = ({ progress }) => {
+  const container = {
+    height: 10,
+    width: '100%',
+    backgroundColor: '#ededed'
+  };
+  const bar = {
+    height: '100%',
+    width: `${progress}%`,
+    backgroundColor: '#4fb0ae',
+    borderRadius: 'inherit'
+  };
+  return (
+    <div style={container}>
+      <div
+        style={bar}
+        role="progressbar"
+        aria-valuenow={progress}
+        aria-valuemin={0}
+        aria-valuemax={100}
+      />
+    </div>
+  );
+};
+
+export const Form = () => {
+  const [inputName, setInputName] = useState('');
+  const [inputRole, setInputRole] = useState('');
+  const [inputEnvironment, setInputEnvironment] = useState('');
+  const [inputExperience, setInputExperience] = useState(null);
+  const [inputTime, setInputTime] = useState(0);
+  const [formStage, setFormStage] = useState(0);
+  const [percent, setPercent] = useState(0);
+
+  const resetForm = () => {
+    setFormStage(0);
+    setInputName('');
+    setInputRole('');
+    setInputEnvironment('');
+    setInputExperience('');
+    setPercent(0);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setFormStage(formStage + 1);
+    const correctPercent = (formStage * 100) / 6;
+    setPercent(Number(correctPercent).toFixed(1));
+  };
+
+  return (
+    <main>
+      {formStage > 0 && <ProgressBar progress={percent} />}
+      <div className="title">
+        <p>
+          {formStage < 5 ? formStage : 5}
+          <span>→</span>
+        </p>
+
+        <h1>Mentorship Sign Up</h1>
+      </div>
+      <div className="image-box">
+        <picture>
+          <source media="(min-width:1042px)" srcSet={headerImgLarge} />
+          <source media="(min-width: 768px)" srcSet={headerImgMedium} />
+          <img src={headerImgSmall} alt="Sign with text: Do something Great" />
+        </picture>
+      </div>
+      {formStage >= 1 && (
+        <button className="reset-button" type="button" onClick={resetForm}>
+          Restart Form
+        </button>
+      )}
+      <form onSubmit={handleSubmit} required>
```

Comment:
So cool that required works well! But is the "required" here needed as well as the required on the different inputs (for example row 105)? Are they both needed? If yes, what are they doing?